### PR TITLE
support conv transpose in script

### PR DESCRIPTION
**Pull request by eellison**

### CODE REVIEW

File: torch/nn/modules/conv.py [hunk 1]
```diff
@@ -460,35 +464,54 @@ def forward(self, input, output_size=None):
...
 
-        return tuple([output_size[d] - min_sizes[d] for d in range(k)])
+    @weak_script_method
+    def _output_padding(self, input, output_size, stride, padding, kernel_size):
+        # type: (Tensor, Optional[List[int]], List[int], List[int], List[int]) -> List[int]
+        if output_size is None:
+            ret = self.to_list(self.output_padding)
+        else:
+            output_size = torch.jit._unwrap_optional(output_size)
+            # output_size = list(output_size)
```
Comment (wanchaol) [inline torch/nn/modules/conv.py]: nit: comment?

File: torch/nn/modules/conv.py [hunk 2]
```diff
@@ -460,35 +464,54 @@ def forward(self, input, output_size=None):
...
-                    "from {} to {} (for an input of {})").format(
-                        output_size, min_sizes, max_sizes, input.size()[2:]))
+    @weak_script_method
+    def to_list(self, elem):
+        # type: (BroadcastingList1[int]) -> List[int]
+        return elem
 
-        return tuple([output_size[d] - min_sizes[d] for d in range(k)])
+    @weak_script_method
+    def _output_padding(self, input, output_size, stride, padding, kernel_size):
```
Comment (wanchaol) [inline torch/nn/modules/conv.py]: is the function signature change necessary?

**Review by wanchaol — APPROVED** [review]
LGTM if CI pass

File: torch/nn/modules/conv.py (hunk 1 above)
Comment (eellison) [reply torch/nn/modules/conv.py]: Yep this needs a fix as well

File: torch/nn/modules/conv.py (hunk 2 above)
Comment (eellison) [reply torch/nn/modules/conv.py]: The alternative is to make lists a constant of the module. By changing the signature, the tuples get converted to lists upon entry of the function, and we can index into the lists with a non-constant value

**Comment by zou3519** [comment]
Test failures look real, please don't land this

**Pull request closed without merging**